sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamBigQuerySqlDialect.java

```diff
@@ -157,8 +160,20 @@ public void unparseCall(
         break;
       case OTHER_FUNCTION:
         String funName = call.getOperator().getName();
-        if (NUMERIC_LITERAL_FUNCTION.equals(funName)) {
-          // self-designed function dealing with the unparsing of ZetaSQL numeric literal
+        if (DOUBLE_POSITIVE_INF_FUNCTION.equals(funName)) {
```

Comment:
If you create a map `DOUBLE_LITERAL_FUNCTIONS`, you can merge the 3 branches here and the 3 functions below. (Similar to what I did with `EXTRACT_FUNCTIONS`)

Reply:
Done. Thanks for pointing it out.